pkg/alertmanager/state_replication.go

```diff
@@ -167,7 +173,22 @@ func (s *state) starting(ctx context.Context) error {
 		}
 	}
 
-	level.Info(s.logger).Log("msg", "state not settled but continuing anyway", "err", err)
+	level.Info(s.logger).Log("msg", "state not settled; trying to read from storage", "err", err)
+
+	// Attempt to read the state from persistent storage instead.
+	storeReadCtx, cancel := context.WithTimeout(ctx, s.storeReadTimeout)
+	defer cancel()
+
+	fullState, err := s.store.GetFullState(storeReadCtx, s.userID)
+	if err == nil {
+		if err = s.mergeFullStates([]*clusterpb.FullState{fullState.State}); err == nil {
+			level.Info(s.logger).Log("msg", "state read from storage; proceeding")
+			return nil
+		}
+	}
+
+	level.Info(s.logger).Log("msg", "failed to read state from storage; continuing anyway", "err", err)
```

Comment:
Assuming we circuit break in case of "object not found" error, then I would raise this to warning:

```suggestion
	level.Warn(s.logger).Log("msg", "failed to read state from storage; continuing anyway", "err", err)
```

Reply:
By circuit break, you mean don't retry any calls to object storage when no object?

Reply:
By circuit break I was meaning a "guard" (if condition then return). There's no retry mechanism here yet (to be discussed if we want it, given retries may also be offered by the bucket client).